core/camel-core/src/main/java/org/apache/camel/impl/JsonApiDataFormat.java

```diff
@@ -0,0 +1,88 @@
+package org.apache.camel.impl;
+
+import com.github.jasminb.jsonapi.JSONAPIDocument;
+import com.github.jasminb.jsonapi.ResourceConverter;
+import org.apache.camel.Exchange;
+import org.apache.camel.spi.DataFormat;
+import org.apache.camel.spi.DataFormatName;
+import org.apache.camel.spi.annotations.Dataformat;
+import org.apache.camel.support.service.ServiceSupport;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * JsonApi {@link org.apache.camel.spi.DataFormat} for marshal/unmarshal
+ */
+@Dataformat("jsonApi")
+public class JsonApiDataFormat extends ServiceSupport implements DataFormat, DataFormatName {
```

Comment:
Except model, this can be moved into another data format in components directory.